### ISSUE TITLE
mp3 being overwritten

### ISSUE THREAD

**Issue by madrover** [jrnewell/spotify-ripper#268]
Hi
Since I have upgraded to 2.10.3 spotify-ripper overwrites existing files that weren't overwritten upon repeated executions previously.

However, this does not happens with all existing files; there are many files that exist already and spotify-ripper skips them as it is expected but others are overwritten on additional executions.

This was working perfectly in previous versions, although I don't know which version did I have before.

Has anything changed regarding this functionality on the latest versions?

Many thanks!

**Comment by orcomma** [comment]
I noticed this too, seems arbitrary

The application states that it's "overwriting partially ripped file"

I'd be interested to know the criteria that defines whether or not the file is partially ripped as the files it is overwriting appear to be full and complete.

Maybe it's a filesize (MB) comparison that's a few KB off?

(I'm encoding 320K MP3 CBR - wondering if that's related - I know LAME is a bit freaky with estimated file sizes and VBR, maybe the same with CBR too?)

**Comment by madrover** [comment]
I don't think that this behavior is really arbitrary, I can see that they're the same files than are overwritten over and over.

I believe that the issue is at https://github.com/jrnewell/spotify-ripper/blob/master/spotify_ripper/utils.py#L454, where spotify-ripper tries to determine if a file is partial or not by comparing their lengths. If the you change the 1500ms value at that line for something bigger then you'll have less false positives.

Ideally, the amount of ms should be configurable IMHO.

**Comment by jrnewell** [comment]
You can turn off this check with ``--partial-check none``.  Can you all confirm that this option fixes it for all of you and it is not due to something else?

The check tries to determine if the file is not fully ripped by comparing the lengths (in time) that Spotify says the file should be to what mutagen says the length of the audio file is.  The audio file has to shorter than what Spotify says by 1.5 seconds.  For some reason, the length Spotify gives is not always correct and Mutagen sometimes has trouble too, so it seems to be getting false positives.

I guess I could change this value to be larger (and make it configurable like @madrover said) or make it a percentage of the song length with a floor.  In addition, I could have ``--partial-check`` be 'none' by default instead of 'weak' - I am not sure how many people want this on by default.  Thoughts?

**Comment by orcomma** [comment]
Thanks James - I'm in the middle of a 2 day playlist rip - when that's complete, I'll see if `--partial-check none` stops the issue.  Out of curiosity what circumstances would cause a partial file to be kept on disk? I'm sure I've seen that, if there is an interruption or error in connection, the ripper deletes the partial file anyway?

**Comment by madrover** [comment]
Personally I'd leave `--partial-check` enabled by default, it's a quite cool function to have enabled by default.
I'd say that 1,5 segs might be a value a bit low and it might be useful to raise the default. I am now doing a couple of tests ripping some lists over existing files so we can have some empirical data that can help tuning this value. So far 124 of 1921 (6'45%) files have been overwritten.
I'll give it a go with 3'' and let you know.

Still, making this value configurable would be nice.

**Comment by madrover** [comment]
ok, so 1,5'' has given me a 6,65% of false positives whereas 3'' has given me 0,57%.
I'd say that 3'' as a default value is still a sane value while given us a much better false positives rate.
